templates/bundle/resources/usr/local/bin/generate_thread_dump.sh

```diff
@@ -48,22 +47,48 @@ function generate_thread_dump {
 	local id=${1}
 	local time=$(date +'%H-%M-%S')
 
-	echo "[Liferay] Generating ${THREAD_DUMPS_DIR}/${date}/thread_dump-${time}-${id}.txt"
+	echo "[Liferay] Generating ${THREAD_DUMPS_DIR}/${date}/thread_dump-${HOSTNAME}-${time}-${id}.txt.gz"
 
 	local thread_dump=$(jattach $(cat "${LIFERAY_PID}") threaddump)
 
-	echo -e "${thread_dump}" > "${THREAD_DUMPS_DIR}/${date}/thread_dump-${time}-${id}.txt"
+	THREAD_DUMP="${THREAD_DUMPS_DIR}/${date}/thread_dump-${HOSTNAME}-${time}-${id}.txt"
+
+	echo -e "${thread_dump}" > "${THREAD_DUMP}"
+
+	gzip "${THREAD_DUMP}"
 }
 
 function main {
 	check_usage "${@}"
 
-	for i in $(seq 1 "${NUMBER_OF_THREAD_DUMPS}")
-	do
-		generate_thread_dump "${i}"
-
-		sleep "${SLEEP}"
-	done
+	if [[ "${LIFERAY_CONTAINER_STATUS_ENABLED}" != "true" ]]
```

Comment:
We shouldn't check for the status, we should attempt to have thread dumps generated whenever it's possible.